tsdb/tsm1/engine_schema.go

```diff
@@ -175,6 +214,14 @@ func (e *Engine) tagValuesPredicate(ctx context.Context, orgBucket, tagKeyBytes
 		}
 
 		for _, iter := range iters {
+			// Always check context before performing a tsm file scan
```

Comment:
I'm don't think we need to check it for every iterator seek, as this operation performance a binary search, so it should execute in O(logN) time. I think every 64 iterations of the outer loop should suffice.